scripts/count-matrix.py

```diff
@@ -1,26 +1,21 @@
 import pandas as pd
 
-def strandedness(units):
+def strandedness(strand_vals):
     strandedness_list = []
-    if "strandedness" in units.columns:
-        for unit in units.itertuples():
-            strand_val = units.loc[(unit.sample, unit.unit), "strandedness"]
-            if pd.isnull(strand_val) or strand_val == "none":
-                strandedness_list.append(1) #non stranded protocol
-            elif strand_val == "yes":
-                strandedness_list.append(2) #3rd column
-            elif strand_val == "reverse":
-                strandedness_list.append(3) #4th column, usually for Illumina truseq
-            else:
-                raise ValueError(("'strandedness' column should be empty or have the " 
+    for strand_val in snakemake.params.strand:
```

Comment:
Instead of creating a list before the loop, I would just zip with `snakemake.params.strand` in the loop, and put the if below inside the loop or as a function for an invididual unit here.

Reply:
I don't really get this, why do you mean with:
>put the if below inside the loop or as a function for an invididual unit here.

The if below is inside the loop so i'm not following you here. If I don't create the list, where I would put the values that I'm returning?

Reply:
```
def get_column(strandedness):
    if pd.isnull(strandedness) or strandedness == "none":
        return 1
    elif strandedness == "yes":
        return 2 #3rd column
    elif strandedness == "reverse":
        return 3 #4th column, usually for Illumina truseq
    else:
        raise ValueError(("'strandedness' column should be empty or have the " 
             "value 'none', 'yes' or 'reverse', instead has the value {}").format(repr(strand_val)))
```

And then:
```
for f, strandedness in zip(snakemake.input, snakemake.params.strand):
    coln = get_column(strandedness)
    ...
```

Reply:
I made the changes, probably you forgot but there is a comprehension list to define `counts` and not a for loop, so I kept it as such and did this: 
```python
 counts = [pd.read_table(f, index_col=0, usecols=[0, get_column(strandedness)], ...
```

Reply:
Yeah, I forgot. Thanks this is great!